spec/lib/middleware/websocket/controller_spec.rb

```diff
@@ -26,17 +26,24 @@
   end
 
   describe '#on_close' do
+    subject { controller.on_close(connection_2) }
+
     before do
       controller.on_open(connection_1)
       controller.on_open(connection_2)
-      controller.on_close(connection_2)
     end
 
     it 'removes the incoming client from the client list' do
+      subject
       clients = controller.clients
 
       expect(clients.length).to eq(1)
       expect(clients.first.connection).to eq(connection_1)
     end
+
+    it 'updates all clients using the write method' do
+      expect(controller.clients[0].connection).to receive(:write)
```

Comment:
Should we test that the correct payload is sent?

Reply:
@jgayfer Would it be more appropriate to test the correct payload in the `Interactor::UpdateAllClients.new.call(clients: @clients)` spec because that's where we are building and sending a payload?

Reply:
I would make the assertion in both places, personally. Currently this test only tells us if we send _something_, which is of little value to us.

The point of writing tests is to test that our application produces the desired side effects, not that we call other code.